roles/ceph-mon/tasks/deploy_monitors.yml

```diff
@@ -0,0 +1,36 @@
+---
+- name: generate monitor initial keyring
+  local_action: shell python -c "import os ; import struct ; import time; import base64 ; key = os.urandom(16) ; header = struct.pack('<hiih',1,int(time.time()),0,len(key)) ; print base64.b64encode(header + key)" | tee fetch/monitor_keyring.conf
+    creates=fetch/monitor_keyring.conf
+  register: monitor_keyring
+  sudo: false
+  when: monitor_secret != 'AQAWqilTCDh7CBAAawXt6kyTgLFCxSvJhTEmuw=='
```

Comment:
This should probably be done as a script or module rather than raw python. Also we shouldn't need to write it out on the filesystem.